### PR TITLE
refactor: Upgrade to mongodb 6.10.0

### PR DESCRIPTION
**Pull request by Moumouls**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: https://github.com/parse-community/parse-server/pull/9297 https://github.com/parse-community/parse-server/pull/9047 https://github.com/parse-community/parse-server/pull/9013

## Tasks

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)
- [x] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [x] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

File: src/Adapters/Files/GridFSBucketAdapter.js [hunk 1]
```diff
@@ -35,8 +35,6 @@ export class GridFSBucketAdapter extends FilesAdapter {
           .substring(0, 32)
         : null;
     const defaultMongoOptions = {
-      useNewUrlParser: true,
```
Comment (mtrezza) [inline src/Adapters/Files/GridFSBucketAdapter.js:38]: Why this refactor?

Comment (Moumouls) [reply src/Adapters/Files/GridFSBucketAdapter.js:38]: yes it fix a mongo warning since it's useless since mongo 4.0 and removed in mongo 7

**Comment by Moumouls** [comment]
failing test seems flaky  @mtrezza

**Comment by Moumouls** [comment]
@mtrezza  can you re run the failed ci ?

**Pull request merged**